doc/tutorials/acrn-dm_QoS.rst

```diff
@@ -0,0 +1,119 @@
+.. _acrn-dm_qos:
+
+Enable QoS based on runC container
+##################################
+ACRN supports Device-Model QoS based on runC container to control
+the SOS resources (CPU, Storage, MEM, NET) by modifying the runC configuration file.
+
+What is QoS
+***********
+Traditionally, Quality of Service (QoS) is the description or measurement
+of the overall performance of a service, such as a `computer network
+<https://en.wikipedia.org/wiki/Computer_network>`_ or a `cloud computing
+<https://en.wikipedia.org/wiki/Cloud_computing>`_ service,
+particularly the performance is seen by the users on the network.
+
+What is runC container
+**********************
+Containers are an abstraction at the application layer that packages code
+and dependencies together. Multiple containers can run on the same machine
+and share the OS kernel with other containers, each running as
+isolated processes in user space. runC, a lightweight universal container runtime,
+is a command-line tool for spawning and running containers according
+to the Open Container Initiative (OCI) specification.
+
+ACRN-DM QoS architecture
+************************
+In ACRN-DM QoS design, we run the ACRN-DM in a runC container environment.
+Every time we start a UOS, we run a runC container at first;
+then launch the ACRN-DM in the started container. The resources
+the ACRN-DM QoS can manage for Device-Model:
+
+- CPU utilization
+- Memory amount/limitation
+- I/O bandwidth
+- Network throughput
+
+.. figure:: images/acrn-dm_qos_architecture.png
+   :align: center
+
+   ACRN-DM QoS architecture
+
+ACRN-QoS CPU utilization example
+********************************
+In runC ``config.json`` we set the CPU resource as below for VM0 and VM1:
+
+.. code-block:: none
+
+    "cpu": {
+       "shares": 1024,
+       "quota": 1000000,
+       "period": 500000,
+       "realtimeRuntime": 950000,
+       "realtimePeriod": 1000000,
+       "cpus": "2-3",
```

Comment:
Are these the CPUs that are given to the VM, or the ones the `acrn-dm` will use (in the SOS)?

I'm wondering how this works because I first thought that this would indicate the `acrn-dm` could use CPU cores 2 and 3. But then I wondered what happens when we take them offline to give them to the VM?

Reply:
Hi Geoffroy,
  The "cpus": "2-3" means you pin the process to the cpu 2-3, but if the cpu offline, it can not set the cpu parameter like this.

Reply:
Our launch scripts typically take all CPUs offline, except the BSP. Is this `config.json` working as it is or does it need some tweaking in the `launch_uos.sh` script?